src/factors/Pose2D.jl

```diff
@@ -38,6 +38,7 @@ end
 
 # Assumes X is a tangent vector
 function (cf::CalcFactor{<:Pose2Pose2})(X, p, q)
+    @assert X isa ProductRepr "Pose2Pose2 expects measurement sample X to be a Manifolds tangent vector, not coordinate or point representation.  Got X=$X"
```

Comment:
Not sure if this is going to help as a point representation is also a `ProductRepr`

Reply:
Sure that's true, will at least help a bit from legacy being coordinates.  I might just leave it in for a while during upgrade to dead reckon tether stuff ... lots of other code there that needs to be updated and assert is helping catch some bugs.  We can remove later if that;s okay.